src/config.rs

```
@@ -223,54 +274,676 @@ fn get_uid_gid(user: &str, group: &str) -> io::Result<(Uid, Gid)> {
     Ok((uid.into(), gid.into()))
 }
 
-pub fn write_configs_to_path(cfgs: &[&WireguardConfig], path: &Path) -> io::Result<()> {
-    let mut file = fs::File::create(path)?;
-    let mut perms = file.metadata()?.permissions();
-    perms.set_mode(0o600);
-    file.set_permissions(perms)?; // Iterate through each configuration and write it to a file
+pub fn write_config_to_path(cfg: &WireguardConfig, path: &Path) -> anyhow::Result<()> {
+    // Helper to delete the file only if it already exists
+    let cleanup = |p: &Path| {
+        if p.exists() {
```

This check seems unnecessary, since removal errors are ignored.